docs: apply suggestions from joss review (#3296)
GEOS-DEV/GEOS: README.md

and applied engineering components of this effort.

Documentation
-------------

> The main documentation for GEOS is hosted [here](https://geosx-geosx.readthedocs-hosted.com/en/latest/?).

Reviewer: Would be a good chance to avoid forms such as `click here` (these are difficult to spot when scanning the page, and not very helpful in other formats, such as screen readers, printed, etc - not a very realistic use case for this particular sentence, but sometimes relevant).

```suggestion
See the [main documentation for GEOS on ReadTheDocs](https://geosx-geosx.readthedocs-hosted.com/en/latest/?).
```
